File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let visited = new Array(n).fill(false);
    let networkCount = 0;

    function dfs(start) {
        visited[start] = true;
        for (let i = 0; i < n; i++) {
            if (!visited[i] && computers[start][i] === 1) {
                dfs(i);
            }
        }
    }

    for (let i = 0; i < n; i++) {
        if (!visited[i]) {
            dfs(i);
            networkCount++;
        }
    }

    return networkCount;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.swift
import Foundation

var visited = [Bool]()
var n = 0

func dfs(_ start: Int, _ computers: [[Int]]) {
    visited[start] = true
    
    for i in 0..<n {        
        if computers[start][i] == 1 && !visited[i] {
            dfs(i, computers)
        }
    }
}

func solution(_ nInput: Int, _ computers: [[Int]]) -> Int {
    n = nInput
    visited = Array(repeating: false, count: n)
    var networkCount = 0    
    for i in 0..<n {
        if !visited[i] {
            dfs(i, computers)
            networkCount += 1
        }
    }
    
    return networkCount
}
